src/table/settings.ini

```diff
@@ -541,6 +541,19 @@ str      = STR_CONFIG_SETTING_MAX_TUNNEL_LENGTH
 strhelp  = STR_CONFIG_SETTING_MAX_TUNNEL_LENGTH_HELPTEXT
 strval   = STR_CONFIG_SETTING_TILE_LENGTH
 
+[SDT_VAR]
+base     = GameSettings
+var      = construction.min_tunnel_depth_under_mine
+type     = SLE_UINT16
+guiflags = SGF_NO_NETWORK
+def      = 1
+min      = 1
+max      = MAX_MAX_HEIGHTLEVEL
```

Comment:
so, to add a new setting, you first add a new saveload version to the end of the enum in saveload.h, just before the line that says
`        SL_MAX_VERSION,                         ///< Highest possible saveload version
`
and to your entry in settings.ini you add a line that says
`from = <insert new enum entry here>`

this way, the game can figure out that a savegame is older, and can insert the new setting at the correct position with its default value upon loading